components/raftstore/src/store/peer_storage.rs

```diff
@@ -457,6 +457,13 @@ where
         }
 
         if find_peer_by_id(&self.region, to).map_or(false, |p| p.is_witness) {
+            if self.applied_index() < request_index {
```

Comment:
check `last_applying_idx` instead of `applied_index`, gen snapshot request goes through apply worker first, so it can make sure getting the latest snapshot.

Reply:
@Connor1996  I don't get your point: as `self.last_applying_idx = last_entry.get_index();`  `request_index` is equal to `switch witness idx` which must be <= `self.last_applying_idx`. So how to use it to check?

BTW, should non-witness set `request_index` to its `last_applying_idx` instead of `switch witness idx`? as we dropped logs between [`switch witness idx`, `last_applying_idx`], if we set `request_index` to its `last_applying_idx`, we don't need `ApplyGap` anymore?

Reply:
`request_index is equal to switch witness idx which must be <= self.last_applying_idx` is not true, leader may get stuck somehow, so it would be slower than the new non-witness.

Reply:
` should non-witness set request_index to its last_applying_idx instead of switch witness idx` no need, as `self.last_applying_idx` is always changing, `ApplyGap` is still needed